File: CDMA.jl
function __init__()
    a = 0
    b = 0
    codeA = Vector{Int64}()
    codeB = Vector{Int64}()
    As = Vector{Int64}()
    Bs = Vector{Int64}()
    Cs = Vector{Int64}()

    print("Enter data A : ")
    A = parse(Int64, readline())
    if A == 0
        A = -1
    end

    print("Enter data B : ")
    B = parse(Int64, readline())
    if B == 0
        B = -1
    end

    println("Enter codeA : ")
    for i = 1:8
         ele = parse(Int64, readline())
         append!(codeA, [ele])
         if codeA[i] == 0
            codeA[i] = -1
         end
    end

    println("Enter codeB : ")
    for i = 1:8
        ele = parse(Int64, readline())
         append!(codeB, [ele])
         if codeB[i] == 0
            codeB[i] = -1
         end
    end

    for i = 1:8
        append!(As, [A * codeA[i]])        
        append!(Bs, [B * codeB[i]])
        append!(Cs, [As[i] + Bs[i]])
        a = a + codeA[i] * Cs[i]
        b = b + codeB[i] * Cs[i]
    end

    println("\n\nCodeA : ", codeA)
    println("CodeB : ", codeB)
    println("As    : ", As)
    println("Bs    : ", Bs)
    println("Cs    : ", Cs)

    if a > 0
        println("\n\nA    : 1")
    else
        println("\n\nA    : 0")
    end

    if b > 0
        println("B    : 1")
    else
        println("B    : 0")
    end
end

__init__()

File: CDMA.rs
use std::io;

fn input() -> i32 {
    let mut input = String::new();
    io::stdin().read_line(&mut input)
               .ok()
               .expect("Cannot read input");

    let input:i32 = input.trim()
                         .parse()
                         .ok()
                         .expect("Wrong data type");
    return input;
}

fn main() {
    let mut a = 0;
    let mut b = 0;
    let mut codea:Vec<i32> = Vec::with_capacity(8);
    let mut codeb:Vec<i32> = Vec::with_capacity(8);
    let mut As:Vec<i32> = Vec::with_capacity(8);
    let mut bs:Vec<i32> = Vec::with_capacity(8);
    let mut cs:Vec<i32> = Vec::with_capacity(8);
    println!("Enter data A : ");
    let mut A:i32 = input();
    if A == 0 {
        A = -1;
    }

    println!("Enter data B: ");
    let mut B:i32 = input();
    if B == 0 {
        B = -1;
    }

    println!("Enter CodeA: ");
    for i in 0..8 {
        let mut ele:i32 = input();
        if ele == 0 {
           ele = -1 
        }
        codea.push(ele);
    }
    println!("codeA : {:?}", codea);

    println!("Enter CodeB: ");
    for i in 0..8 {
        let mut ele:i32 = input();
        if ele == 0 {
           ele = -1 
        }
        codeb.push(ele);
    }
    println!("codeB : {:?}", codeb);

    for i in 0..8 {
        As.push(A*codea[i]);
        bs.push(B*codeb[i]);
        cs.push(As[i]+bs[i]);
        a = a + codea[i]*cs[i];
        b = b + codeb[i]*cs[i];
    }
    println!("CodeA : {:?}, CodeB: {:?}, As : {:?}, Bs: {:?}, Cs: {:?}", codea, codeb, As, bs, cs);

    if a > 0 {
        println!("\n\nA : 1");
    }
    else {
        println!("\n\nA : 0");
    }

    if b > 0 {
        println!("B : 1");
    }
    else {
        println!("B : 0");
    }
}